packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -301,11 +299,10 @@ export class ApolloServerBase {
     }
 
     if (gateway && subscriptions !== false) {
-      if (typeof subscriptions === 'undefined') {
-        console.warn('Subscriptions disabled when operating as a gateway.');
-      } else {
-        throw new Error('Cannot define both `subscriptions` and `gateway`.');
-      }
+      // TODO: this could be handled by adjusting the typings to keep gateway configs and non-gateway configs seprate.
+      throw new Error(
+        "Cannot define both `subscriptions` and `gateway`. Set `subscriptions: false` in Apollo Server's config to use gateway.",
+      );
```

Comment:
Same note as above about TypeScript.  As much as we love it, not everyone uses it!